classes/class-wpcom-liveblog-entry-extend-feature-hashtags.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+/**
+ * Class WPCOM_Liveblog_Entry_Extend_Feature_Hashtags
+ *
+ * The base class for autocomplete features.
+ */
+class WPCOM_Liveblog_Entry_Extend_Feature_Hashtags extends WPCOM_Liveblog_Entry_Extend_Feature {
+
+	/**
+	 * The taxonomy name.
+	 *
+	 * @var string
+	 */
+	protected static $taxonomy = 'hashtags';
+
+	/**
+	 * The class prefix.
+	 *
+	 * @var string
+	 */
+	protected $class_prefix = 'term-';
+
+	/**
+	 * The character prefixes.
+	 *
+	 * @var array
+	 */
+	protected $prefixes = array( '#', '\x{ff03}' );
+
+	/**
+	 * Called by WPCOM_Liveblog_Entry_Extend::load()
+	 *
+	 * @return void
+	 */
+	public function load() {
+		$this->class_prefix = apply_filters( 'liveblog_hashtag_class', $this->class_prefix );
+
+		$this->revert_regex = implode( '', array(
+			preg_quote( '<span class="liveblog-hash ', '~' ),
+			preg_quote( $this->class_prefix, '~' ),
+			'([^"]+)',
+			preg_quote( '">', '~' ),
+			'\1',
+			preg_quote( '</span>', '~' ),
+		) );
+
+		$this->revert_regex = apply_filters( 'liveblog_hashtag_revert_regex', $this->revert_regex );
+
+		add_filter( 'comment_class',          array( $this, 'add_term_class_to_entry' ), 10, 3 );
+		add_action( 'init',                   array( $this, 'add_hashtag_taxonomy') );
+		add_action( 'wp_ajax_liveblog_terms', array( $this, 'ajax_terms') );
+	}
+
+	/**
+	 * Gets the autocomplete config.
+	 *
+	 * @param array $config
+	 * @return array
+	 */
+	public function get_config( $config ) {
+		$config[] = apply_filters( 'liveblog_hashtag_config', array(
+			'type'        => 'ajax',
+			'cache'       => 1000 * 60,
+			'regex'       => '#(\w*)$',
+			'url'         => admin_url( 'admin-ajax.php' ) .'?action=liveblog_terms',
+			'replacement' => '#${term}',
+		) );
+
+		return $config;
+	}
+
+	/**
+	 * Filters the input.
+	 *
+	 * @param mixed $entry
+	 * @return mixed
+	 */
+	public function filter( $entry ) {
+		$entry['content'] = preg_replace_callback(
+			$this->get_regex(),
+			array( $this, 'preg_replace_callback' ),
+			$entry['content']
+		);
+
+		return $entry;
+	}
+
+	/**
+	 * The preg replace callback for the filter.
+	 *
+	 * @param array $match
+	 * @return string
+	 */
+	public function preg_replace_callback( $match ) {
+		$term = iconv( 'UTF-8', 'ASCII//TRANSLIT', $match[2] );
+
+		if ( ! term_exists( $term, self::$taxonomy ) ) {
+			wp_insert_term( $term, self::$taxonomy );
+		}
+
+		return str_replace(
+			$match[1],
+			'<span class="liveblog-hash '.$this->class_prefix.$term.'">'.$term.'</span>',
```

Comment:
Hey @imsimond,

With my [last comment](https://github.com/Automattic/liveblog/pull/190#discussion_r34900122) — they will now be using slugs :-)